app/lib/account/consent_backend.dart

```diff
@@ -185,7 +198,8 @@ class ConsentBackend {
       InvalidInputException.check(c.userIdOfConsent == user.userId,
           'This invitation is not for the user account currently logged in.');
     }
-    if (c.email != null) {
+    final permitsDifferentUser = c.kind == ConsentKind.publisherContact;
```

Comment:
`_actions[c.kind].permitConfirmationByUserWithOtherEmail` and then add `ConsentAction.permitConfirmationByUserWithOtherEmail` (and document what this property means).

I think this is much safer than special casing on `c.kind`.